### PR TITLE
[NCL-2039] add sorting of artifacts in report tree

### PR DESCRIPTION
**Pull request by stanlyDoge**

### CODE REVIEW

File: communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java [hunk 1]
```diff
@@ -36,9 +40,9 @@ public GAVDependencyTree getDependencyTree(Set<DependencyRelationship> rels, GAV
 
         for(DependencyRelationship rel : rels){
             GAV gav = GalleyWrapper.generateGAV(rel.getDeclaring());
-            Set<DependencyRelationship> deps = byGav.computeIfAbsent(gav, (k) -> new HashSet<>());
+            Set<DependencyRelationship> deps = byGav.computeIfAbsent(gav, (k) -> new ConcurrentSkipListSet<>(new DependencyRelationshipComparator()));
             deps.add(rel);
-        }
+          }
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java]: why the added spaces?

File: communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java [hunk 2]
```diff
@@ -36,9 +40,9 @@ public GAVDependencyTree getDependencyTree(Set<DependencyRelationship> rels, GAV
 
         for(DependencyRelationship rel : rels){
             GAV gav = GalleyWrapper.generateGAV(rel.getDeclaring());
-            Set<DependencyRelationship> deps = byGav.computeIfAbsent(gav, (k) -> new HashSet<>());
+            Set<DependencyRelationship> deps = byGav.computeIfAbsent(gav, (k) -> new ConcurrentSkipListSet<>(new DependencyRelationshipComparator()));
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java]: The TreeSet is probably better as this is not parallel.

File: testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (janinko) [inline testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java]: Remove the "license" header please

File: testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        gav1 = new GAV("com.megginson.sax","xml-writer","0.2");
+        gav2 = new GAV("aom.megginson.sax","xml-writer","0.2");
+        assertTrue(gav1.compareTo(gav2) > 0);
+        
+        //random retezec
+        gav1 = new GAV("junit","junit","3.8.1");
+        gav2 = new GAV("com.thoughtworks.xstream","xstream","1.4.2");
+        assertTrue(gav1.compareTo(gav2) > 0);
+        
+        //random retezec
```
Comment (janinko) [inline testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java]: Please write in english.

File: testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        GAV gav4 = new GAV("org.testng", "testng", "5.10");
+        GAV gav5 = new GAV("javax.ws.rs", "javax.ws.rs-api", "2.0");
+        TreeSet<GAV> set = new TreeSet<>();
+        set.add(gav1);
+        set.add(gav2);
+        set.add(gav3);
+        set.add(gav4);
+        set.add(gav5);
+        for(GAV g : set){
+            System.out.println(g);
```
Comment (janinko) [inline testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java]: Can you rework this test so it actually test the order and doesn't print it out?

File: testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.da.model.rest.GAV;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ *
+ * @author sknot
```
Comment (janinko) [inline testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java]: Maybe add your full name and email.

File: communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java [hunk 7]
```diff
@@ -108,7 +110,7 @@ private void walkDependencies(PathNode root, Map<GAV, Set<DependencyRelationship
 
         private final GAV gav;
 
-        private final Set<PathNode> childs = new HashSet<>();
+        private final Set<PathNode> childs = new TreeSet<>(new PathNodeComparator());
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java]: Didn't you want to try lambdas? Or you can change PathNode to implement Comparable. Or you can leave it as is. :)

Comment (stanlyDoge) [inline communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java]: The last option is the best one :)

File: testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ *
+ * @author Stanislav Knot <knot.stanislav@gmail.com>
+ */
+public class GAVComparatorRemoteTest {
+    
+
+    @Test
+    public void test() {
+        //stejne
```
Comment (janinko) [inline testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java]: Please write in english

File: testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.da.model.rest.GAV;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ *
+ * @author Stanislav Knot <knot.stanislav@gmail.com>
```
Comment (janinko) [inline testsuite/src/test/java/org/jboss/da/test/server/communication/GAVComparatorRemoteTest.java]: Your work email would be better.

**Comment by janinko** [comment]
:+1:

**Comment by jomrazek** [comment]
:+1:

**Comment by janinko** [comment]
retest this please

**Comment by janinko** [comment]
Wrong formating

**Pull request merged**